### PR TITLE
BUG: Index.sort_values with key

### LINKED ISSUE #52764: BUG: Index.sort_values with key
`Index.sort_values` does 

```
idx = ensure_key_mapped(self, key)
_as = nargsort(
    items=idx, ascending=ascending, na_position=na_position, key=key
)
```

But when key is not None, nargsort does the ensure_key_mapped call again.  And ensure_key_mapped is not idempotent.

If I remove the `key=key` from the nargsort call here, the only test that fails is `TestRangeIndex::test_sort_values_key`

```
sort_order = {8: 2, 6: 0, 4: 8, 2: 10, 0: 12}
values = RangeIndex(0, 10, 2)
result = values.sort_values(key=lambda x: x.map(sort_order))
expected = Index([4, 8, 6, 0, 2], dtype="int64")
tm.assert_index_equal(result, expected, check_exact=True)
```

but it isn't obvious to me `expected` here is correct.  If we do `pd.Series(values).sort_values(key=lambda x: x.map(sort_order))` we get a different ordering.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #52764 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -624,11 +624,13 @@ Other
 - Bug in :meth:`DataFrame.reindex` with a ``fill_value`` that should be inferred with a :class:`ExtensionDtype` incorrectly inferring ``object`` dtype (:issue:`52586`)
 - Bug in :meth:`DataFrame.shift` and :meth:`Series.shift` and :meth:`DataFrameGroupBy.shift` when passing both "freq" and "fill_value" silently ignoring "fill_value" instead of raising ``ValueError`` (:issue:`53832`)
 - Bug in :meth:`DataFrame.shift` with ``axis=1`` on a :class:`DataFrame` with a single :class:`ExtensionDtype` column giving incorrect results (:issue:`53832`)
+- Bug in :meth:`Index.sort_values` whena  ``key`` is passed (:issue:`52764`)
```
Comment (rhshadrach) [inline doc/source/whatsnew/v2.1.0.rst:627]: Nit: when a

Comment (rhshadrach) [reply doc/source/whatsnew/v2.1.0.rst:627]: Also - move to the Indexing section?

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v2.1.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v2.1.0.rst:627]: Good catch, will update.

As to the section, its pretty subjective but I usually think of "indexing" as about Series/DataFrame/loc.iloc getitem/setitem methods

Comment (jbrockmendel) [reply doc/source/whatsnew/v2.1.0.rst]: updated+green

**Review by rhshadrach — APPROVED** [review]
lgtm

**Pull request merged**